server/src/main/java/org/elasticsearch/common/util/concurrent/EsRejectedExecutionHandler.java

```diff
@@ -26,6 +29,14 @@ public long rejected() {
 
     protected void incrementRejections() {
         rejected.inc();
+        if (telemetry_rejected != null) {
+            telemetry_rejected.increment();
+        }
+    }
+
+    public void registerCounter(MeterRegistry meterRegistry, String prefix, String name) {
+        telemetry_rejected = meterRegistry.registerDoubleCounter(prefix + "rejected", "number of rejected threads for " + name, "count");
```

Comment:
can we use 'total' suffix here?

Reply:
btw why double counter not a long counter?

Reply:
I am not too sure why I took double instead of long there, changed to Long now in 1a0b1defba83ade4b673cf9af21fa09161397f50